napari/_vispy/canvas.py

```diff
@@ -1,36 +1,90 @@
 """VispyCanvas class.
 """
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
 from weakref import WeakSet
 
-from vispy.scene import SceneCanvas, Widget
+import numpy as np
+from vispy.scene import SceneCanvas as SceneCanvas_
+from vispy.scene import Widget
 
+from napari._vispy import VispyCamera
 from napari._vispy.utils.gl import get_max_texture_sizes
+from napari.utils._proxies import ReadOnlyWrapper
 from napari.utils.colormaps.standardize_color import transform_color
+from napari.utils.interactions import (
+    mouse_double_click_callbacks,
+    mouse_move_callbacks,
+    mouse_press_callbacks,
+    mouse_release_callbacks,
+    mouse_wheel_callbacks,
+)
+
+if TYPE_CHECKING:
+    from napari.components import ViewerModel
 
 
-class VispyCanvas(SceneCanvas):
+class SceneCanvas(SceneCanvas_):
+    """Vispy SceneCanvas used to allow for ignoring mouse wheel events with modifiers."""
+
+    def _process_mouse_event(self, event):
+        """Ignore mouse wheel events which have modifiers."""
+        if event.type == 'mouse_wheel' and len(event.modifiers) > 0:
+            return
+        super()._process_mouse_event(event)
+
+
+class VispyCanvas:
     """SceneCanvas for our QtViewer class.
 
     Add two features to SceneCanvas. Ignore mousewheel events with
     modifiers, and get the max texture size in __init__().
```

Comment:
I think this class is responsible for much more now, so please update the docstring to reflect that.

Reply:
Ah I overlooked that, you are right. I will adjust the docstring accordingly